instrumentation/base/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -107,6 +107,22 @@ def instrumentation_version(instrumentation_version = nil)
           end
         end
 
+        # Optionally set the schema_url of this instrumentation. If not explicitly set,
+        # the schema_url will default to remaining unset by the instrumentation, and
+        # will use the applicaiton wide (resource) schema_url. If an application wide
+        # (resource) schema_url cannot be found, it defaults to remaining unset and
+        # is set by the OTLP exporter using the schema of the exporter.
+        # TODO: clarify this^, is this correct? where does app wide schema url get used?
```

Comment:
this sounds correct to me, but when i tried to move this into `attr_writer  :instrumentation_schema_url` couldn't seem to get it to work well the tests. I went down a rabbit hole of trying to understand why and couldn't come up with a clear answer, will try to look at this with fresh eyes. I think our pattern of instantiating an instrumentation instance doesn't play nicely with this approach.

other than this everything else should be resolved afaik

Reply:
i'm kinda stumped here. Changing to an `attr_writer` or `attr_accessor` causes a slight pattern change in the way an instrumentation author defines an `instrumentation_schema_url` (`instrumentation_schema_url = <schema_url>`) vs how they define `instrumentation_version` (`instrumentation_version(<version>)`) or `instrumentation_name`. (`instrumentation_name(<name>)` in their instrumentation Class. Updating test classes to account for this, it still seems to be problematic when we create a [new `instance`](https://github.com/open-telemetry/opentelemetry-ruby/pull/979/files#diff-7379de806670e7a913ebbe1d8e27945a6367cc83c94d79e55434de15cdd16732R167). But defining the method instead of using `attr_` makes this problem go away. Maybe `attr_` doesn't define the getters and setters at the appropriate time, whereas defining it explicitly does? 🤷